src/circuit/gadget/ecc/chip/mul_fixed/base_field_elem.rs

```diff
@@ -0,0 +1,645 @@
+                    &mul_b,
+                    &acc,
+                    offset + NUM_WINDOWS,
+                    &mut region,
+                )?;
+
+                #[cfg(test)]
+                // Check that the correct multiple is obtained.
+                {
+                    use group::Curve;
+
+                    let base: super::OrchardFixedBases = base.into();
+                    let scalar = &scalar
+                        .base_field_elem()
+                        .value()
+                        .map(|scalar| pallas::Scalar::from_bytes(&scalar.to_bytes()).unwrap());
+                    let real_mul = scalar.map(|scalar| base.generator() * scalar);
+                    let result = result.point();
+
+                    if let (Some(real_mul), Some(result)) = (real_mul, result) {
+                        assert_eq!(real_mul.to_affine(), result);
+                    }
+                }
+
+                Ok((result, scalar))
+            },
+        )?;
+
+        // We want to enforce canonicity of a 255-bit base field element, α.
+        // That is, we want to check that 0 ≤ α < p, where p is Pallas base
+        // field modulus p = 2^254 + t_p
+        //                 = 2^254 + 45560315531419706090280762371685220353.
+        // Note that t_p < 2^130.
+        //
+        // α has been decomposed into three pieces,
+        //            α = α_0 (252 bits)  || α_1 (2 bits) || α_2 (1 bit).
+        //
+        // If the MSB α_2 = 1, then:
+        //      - α_2 = 1 => α_1 = 0, and
+        //      - α_2 = 1 => α_0 < t_p. To enforce this:
+        //      - α_2 = 1 => 0 ≤ α_0 < 2^130
+        //                => 13 ten-bit lookups of α_0
+        //      - α_2 = 1 => 0 ≤ α_0 + 2^130 - t_p < 2^130
+        //                => 13 ten-bit lookups of α_0 + 2^130 - t_p
+        //                => z_13_alpha_0_prime = 0
+        //
+        let (alpha, running_sum) = (scalar.base_field_elem, &scalar.running_sum);
+        let z_84_alpha = running_sum[83];
+        let z_85_alpha = running_sum[84];
+
+        let (alpha_0, z_13_alpha_0) = {
+            // α_0 = α - z_84_alpha * 2^252
+            let alpha_0 = alpha
+                .value()
+                .zip(z_84_alpha.value())
+                .map(|(alpha, z_84_alpha)| {
+                    let two_pow_252 = pallas::Base::from_u128(1 << 126).square();
+                    alpha - z_84_alpha * two_pow_252
+                });
+
+            let (alpha_0, zs) = self.lookup_config.witness_check(
+                layouter.namespace(|| "Lookup range check alpha + 2^130 - t_p"),
+                alpha_0,
+                13,
+                false,
+            )?;
+
+            (alpha_0, zs[13])
+        };
+
+        let (alpha_0_prime, z_13_alpha_0_prime) = {
+            // alpha_0_prime = alpha + 2^130 - t_p.
+            let alpha_0_prime = alpha_0.value().map(|alpha_0| {
+                let two_pow_130 = pallas::Base::from_u128(65).square();
```

Comment:
Ah, this is probably why tests pass: the bug is duplicated here.
```suggestion
                let two_pow_130 = pallas::Base::from_u128(1 << 65).square();
```